contracts/pool/concentrated/ConcentratedLiquidityPool.sol

```diff
@@ -123,10 +124,11 @@ contract ConcentratedLiquidityPool is IPool {
             _deployData,
             (address, address, uint24, uint160, uint24)
         );
-
+        if (TickMath.MIN_SQRT_RATIO <= price && price <= TickMath.MAX_SQRT_RATIO) revert BadPrice();
         if (_token0 == address(0)) revert ZeroAddress();
         if (_token0 == address(this)) revert InvalidToken();
         if (_token1 == address(this)) revert InvalidToken();
```

Comment:
If you're using CREATE2, these should be impossible. Not worth the gas IMO. We have all these worthless checks but not token0 < token1

Reply:
This check is important and is not prevented by create2, can you throw some light on how create2 prevents this, pricing check?

Reply:
sha256

Reply:
The create2 takes the hash of our deployment data as salt, not sure which check are you referring to here, you think is irrelevant. Anyone can just slightly modify the price or any deployment parameter to deploy a new pair.

Reply:
If you can break keccak then you can make a few trillion dollars

Reply:
generally, given an output, you cannot generate an input. keccak is a one-way function.

Reply:
Yes, That's correct, but from the data that the keccak256 is calculated from is already known. It is the deployment data, which is hashed as used as salt.

Reply:
and token0, token1 are part of deployment data

Reply:
Yes, ```_deployData = abi.encode(tokenA, tokenB, swapFee, price, tickSpacing);```

Reply:
> Anyone can just slightly modify the price or any deployment parameter to deploy a new pair.

At p = 2^-160, we would need to do 10^31 keccaks per second on average to expect one collision before the death of the sun. All bitcoin mining currently achieves 10^17. So if we utilized the energy of 100 quadrillion bitcoin networks for 5 billion years, we could expect this to happen.

Reply:
What is the exact issue that you want to be resolved here, can you please explain?

Reply:
I want to skip these address(this) checks.

Reply:
Haha, I get it now, you highlighted the wrong line, which confused me.

Reply:
@wjmelements Made the changes requested. Thank you!